[Ready for review] Open science chapter- low level reviews

r-j-arnold opened the pull request:
The pull request for discussion high level changes to the chapter (i.e. changes on the scale of a paragraph or more such as restructuring and adding material) has been completed and merged to the open science branch. This pull request is for discussing low level changes, such as rewording sentences, typos and the like.

pherterich commented on chapters/open_science.md:
```diff
@@ -1,6 +1,6 @@
-## Summary
+## Summary 
 
 Open Science aims to transform science by making research more reproducible, transparent, re-usable, collaborative, accountable and closer to society. It is about the way research is carried out and disseminated by digital tools. The OECD defines Open Science as:  “to make the primary outputs of publicly funded research results – publications and the research data – publicly accessible in digital ...
```
Could we make this a proper sentence, e.g. "The OECD defines Open Science as *the practice* “to make the primary outputs of publicly funded research results – publications and the research data – publicly accessible in digital format with no or minimal restriction” or something? Also, which of the references in the bibliography is this quote from? Could you make that connection?

r-j-arnold replied:
I've done this and added a link to the source of the quote.

r-j-arnold commented on .github/ISSUE_TEMPLATE.md:
```diff
@@ -1,16 +1,49 @@
-To avoid that others have to read through the full thread of comments, update the initial issue with important updates (e.g. decisions taken) regularly so that all the important information is available at one glance.\ 
-We recommend doing this using
 
 ---
```
test review

r-j-arnold commented:
> This might be a really dumb question, but how to I add comments to individual lines? GitHub doesn't offer me the option to do that except for the part that has changed and if I just start adding comments, this will turn into a complete mess. Suggestions anyone? Am I missing some obvious GitHub functionality again?

@pherterich When I go onto the "files changed" tab and then hover my mouse over the actual code a little blue + symbol appears on the left of whatever line I'm hovering over. If I click that I can then comment on that specific line. Do you see the same thing (it may be different since I made this pull request.

pherterich commented:
@r-j-arnold I see it for the first 8 lines, basically the part that you made a change to last. The rest of the file is grayed out and doesn't offer that functionality and I have no idea if I could make that happen somehow or how else to leave comments in a sensible way.

r-j-arnold commented:
@pherterich Looking online I've found people saying the same as you; that they can't comment on those lines and it would be a damn useful if they could. Unfortunately the complaints date from 2013 up to 2018, so it doesn't look like there is a way or that GitHub intends to make one any time soon.

Do you see the line numbers in either the diff or if you look at the file itself? Referencing those like

27: Misspelled "break" in "can brek the implementation"

or something like that could work

pherterich commented:
I see line numbers, but just hoped there would be a smarter way. Will do a proper review then next week creating one big comment with all the issues.

r-j-arnold commented:
That's great, thanks @pherterich

LouiseABowler commented:
@r-j-arnold, do you want to make this a PR into master rather than open_science? (Or go ahead and merge this one, then do another PR from the-turing-way/open_science into master?) That should display the entire open_science chapter in the diff so we can give you comments by line rather than a general review.

r-j-arnold commented:
@LouiseABowler I'm hesitant to do that because the idea of having separate branch is that all the development and changes happen here and it's only the finished(ish) project that gets submitted to master. You make a good point it would make reviews easier, I suggest we talk about moving to that model in the next meeting (and if so update the contributing guidelines/ways of working), but for now keep it like this.

pherterich commented:
General comment: If we could follow good practice and cite documents using their DOI where available that would be really good and show good practice. I don't really understand what the documents in the bibliography support, so if that could be turned into proper citations in text, I think that would be helpful. There will be tidying for consistency in the end, but pointers which parts of the chapter have been taken from the resources listed in the bibliography would be really useful as otherwise it's really hard for someone to edit this and attribute sources and licences correctly.

r-j-arnold commented:
> General comment: If we could follow good practice and cite documents using their DOI where available that would be really good and show good practice. I don't really understand what the documents in the bibliography support, so if that could be turned into proper citations in text, I think that would be helpful. There will be tidying for consistency in the end, but pointers which parts of the chapter have been taken from the resources listed in the bibliography would be really useful as otherwise it's really hard for someone to edit this and attribute sources and licences correctly.

The issue I'm having with the references is that when I incorporate a resource I usually need to put chunks of it all over the chapter to combine resources and fit them in with the overall flow of the chapter. These chunks are sometimes contained paragraphs but are usually sentences, half-sentences, and can sometimes be as small as few-word phrases and it's just not practical to have up to 8 citations per resource.

The way I've done it sucks, I fully acknowledge that, and it needs changing, but I don't think the standard way of citing things as *blah blah blah* (citation) is going to work in this scenario.

One thing I had previously is mini bibliographies in each subsection, so e.g all the open hardware resources went at the end of that subsection. Maybe I could restore that?

Good point about DOIs, I'll have a look through, and add them in for any resources that have one.

pherterich commented:
> The issue I'm having with the references is that when I incorporate a resource I usually need to put chunks of it all over the chapter to combine resources and fit them in with the overall flow of the chapter. These chunks are sometimes contained paragraphs but are usually sentences, half-sentences, and can sometimes be as small as few-word phrases and it's just not practical to have up to 8 citations per resource.

Well, unless the source is CC0, a different approach might be more practical but will be either plagiarism and/or a breach of the licence. Having them closer to each section will already help, then someone can go through it in the end and make sure everything is attributed reasonably well.

pherterich commented:
- [ ] line 13: -> Open Data - sharing well documented research data openly for re-use
- [ ] line 41 -> open access articles are cited more often
- [ ] line 52: maybe call this section Open (research) data just to be clear. You talk about scientific data a lot and I'm not sure if that might exclude text corpora etc. @rosiehigman suggestions for rephrasing to make it inclusive? It might also be worth adding a note to open government data here as a movement and how that makes research already easier in some fields, 
- [ ] line 61: "without financial, legal, or technical barriers" that's debatable. Is data under a CC-BY-NC licence still open? I'd say so.
- [ ] line 63 to 110 will be covered in the RDM chapter, so you can replace that with a signpost
- [ ] line 114: under permissive licenses -> licences
- [ ] line 122: Open hardware also makes science more reproducible ~~reproduceable~~ as researchers looking to verify results can do the same thing.
- [ ] line 126: you use open-source in the headline and then open source in e.g. line 128. Pick one and use it consistently (preference is without - as that's what I'm more used to, but in the end it doesn't really matter)
- [ ] line 131: license -> licence; This grants legal permission to anyone to re-use, modify and distribute your designs and hardware *according to the terms stated*.
- [ ] line 143: what's an STL? you haven't introduced that acronym before. 
However, note that auxiliary design files are never allowed as substitutes for original design files. -> allowed is a strong word, is that really what you mean? or is it more recommended?

pherterich commented:
- [ ] line 181: license -> licence. Would "distribute your project *code* for any purpose." be more accurate?
- [ ] line 183: out of interest: is there a source for "would, at the very least, have required a budget of tens of millions of dollars" ?
- [ ] line 190: practice -> practise
- [ ] line 193: missing . at the end of the sentence
- [ ] line 197: is this a sub point of line 191? I would potentially move this around a bit.
- [ ] line 199: are there studies that support this for code and software? Sources would be good here...
- [ ] line 213: links to official web pages for these projects would be good
- [ ] line 217: for code, you could explicitly promote https://guides.github.com/activities/citable-code/ or the figshare equivalent here
- [ ] line 219: license -> licence
- [ ] line 229: should project be project code here? Assuming that you just want to run the code and not take over project management?
- [ ] line 236 to 255 could probably be covered in the collaborating through GitHub chapter and be taken out here
- [ ] line 316 to 355: not sure how you would like me to address the rewrite of the OA section, shall I do a PR to this file?

r-j-arnold commented:
> * [ ]  line 13: -> Open Data - sharing well documented research data openly for re-use
Changed to "documenting and sharing research data openly for re-use" since I think "sharing well documented research data" makes it sound like we *only* advocate sharing data that is already well documented.

> * [ ]  line 41 -> open access articles are cited more often
> * [ ]  line 52: maybe call this section Open (research) data just to be clear. You talk about scientific data a lot and I'm not sure if that might exclude text corpora etc. @rosiehigman suggestions for rephrasing to make it inclusive? It might also be worth adding a note to open government data here as a movement and how that makes research already easier in some fields,
Good point about exclusivity. "Science" changed to "research" throughout except right at the start since (as it stands) this is called the open science chapter. Suggest once this is in master we delete the open science branch, start a new one called open research and retitle the chapter. 

Also in the first parapgraph on the open data section I've modified a sentence to "This lack of data sharing is an obstacle to international research (be it academic, governmental or commercial) for two main reasons:" to make it clear it doesn't have to be scientific data.

> * [ ]  line 61: "without financial, legal, or technical barriers" that's debatable. Is data under a CC-BY-NC licence still open? I'd say so.
I've changed this to "with *minimal*" rather than "without" 

> * [ ]  line 63 to 110 will be covered in the RDM chapter, so you can replace that with a signpost
I'll do that once RDM is ready to go to master.

> * [ ]  line 114: under permissive licenses -> licences
Just changed it to "available to everyone" rather than "available to everyone under permissive licenses"
> * [ ]  line 122: Open hardware also makes science more reproducible reproduceable as researchers looking to verify results can do the same thing.
> * [ ]  line 126: you use open-source in the headline and then open source in e.g. line 128. Pick one and use it consistently (preference is without - as that's what I'm more used to, but in the end it doesn't really matter)
> * [ ]  line 131: license -> licence; This grants legal permission to anyone to re-use, modify and distribute your designs and hardware _according to the terms stated_.
> * [ ]  line 143: what's an STL? you haven't introduced that acronym before.
Deleted that example, as not necessary. 

>   However, note that auxiliary design files are never allowed as substitutes for original design files. -> allowed is a strong word, is that really what you mean? or is it more recommended?
Changed to recommended.

pherterich commented:
- [ ] line 359: "Open notebook science" sounds a bit like a science in itself. I think this is more commonly referred to as the sharing of Electronic Lab Notebooks (ELNs). They don't have to be online, I think they can be local on machines and they definitely do not have to be open from the start.
- [ ] line 369: I think it's worth saying something about "scholarship" being more inclusive to cover arts and humanities and just all kinds of research and not necessarily "hard science" only
- [ ] line 420: license -> licence
- [ ] line 426: license -> licence

r-j-arnold commented:
> * [ ]  line 181: license -> licence. Would "distribute your project _code_ for any purpose." be more accurate?

Not really, for example documentation isn't code but it goes along with it as part of the project and is just as open.

> * [ ]  line 183: out of interest: is there a source for "would, at the very least, have required a budget of tens of millions of dollars" ?

Not that they give, but given the amount of highly-skilled hours that go into packages like that (or numpy, scipy etc) it seems pretty unarguable. 

> * [ ]  line 190: practice -> practise

I believe practise is the american spelling, correct me if I'm wrong

> * [ ]  line 193: missing . at the end of the sentence
> * [ ]  line 197: is this a sub point of line 191? I would potentially move this around a bit.

It kinds of fits into both. I struggled with how to structure this section and this was the best I could come up with, but I'm open to restructuring. 

> * [ ]  line 199: are there studies that support this for code and software? Sources would be good here...

Changed to "Making your work openly available for re-use makes it easier for others to incorporate into their own research. If you make your software citeable, for example via a DOI this can increase your citations."

> * [ ]  line 213: links to official web pages for these projects would be good
> * [ ]  line 217: for code, you could explicitly promote https://guides.github.com/activities/citable-code/ or the figshare equivalent here

Wherever possible I'm trying to make sure the chapter covers everything rather than asking readers to go elsewhere for info. Is there any specific info there you wanted put in?

> * [ ]  line 219: license -> licence
> * [ ]  line 229: should project be project code here? Assuming that you just want to run the code and not take over project management?
> * [ ]  line 236 to 255 could probably be covered in the collaborating through GitHub chapter and be taken out here

It's also in the github chapter and I point people to that chapter for more details, but I felt it was necessary to include the minimum a reader would need for working open source in the open source section

> * [ ]  line 316 to 355: not sure how you would like me to address the rewrite of the OA section, shall I do a PR to this file?

Ideally do a PR to master after this is merged in to reduce the likelihood of merge conflicts.

r-j-arnold commented:
> * [ ]  line 359: "Open notebook science" sounds a bit like a science in itself. I think this is more commonly referred to as the sharing of Electronic Lab Notebooks (ELNs). They don't have to be online, I think they can be local on machines and they definitely do not have to be open from the start.
> * [ ]  line 369: I think it's worth saying something about "scholarship" being more inclusive to cover arts and humanities and just all kinds of research and not necessarily "hard science" only

At the moment the chapter's titled open science, as in an above reply I suggest once this open_science branch is merged to master we make one called open_research sand reorient the chapter there. For the moment I'll leave this as it is, but you're right it should be changed and I'll do it then.

> * [ ]  line 420: license -> licence
> * [ ]  line 426: license -> licence

pherterich commented:
> > * [ ]   line 190: practice -> practise
> 
> I believe practise is the american spelling, correct me if I'm wrong

You're right on that one. Reading through a lot of text usually gets me confused. As a non native speaker, I get lost here and there.

LouiseABowler commented:
Most of my comments from the first third (or so) of this chapter were in the HackMD, so I'll start from approximately where I left off. I'll try to do one comment per section for now.

- [ ] Move prerequisites up to the top of the chapter.

r-j-arnold commented:
@LouiseABowler One comment per sections sounds like a perfect way of breaking it down into manageable chunks

LouiseABowler commented:
Open data

Quite a few of these are rephrasing suggestions; feel free to ignore them if you'd rather leave the text as it is. 

- [ ] Line 61: "permitting any user" -> "and any user is permitted"
- [ ] Line 62: Could "Open by default" be put in quotes?
- [ ] Line 62: "Researchers need to justify data..." -> "If access to data is restricted, for instance due to security reasons, the justification for this should be made clear."
- [ ] Line 69: "As much as possible researchers" -> "Whenever possible, researchers"
- [ ] Line 75: Add a comma after "For example"
- [ ] Line 75: "what the columns/rows contain" -> "the contents of the rows and columns"
- [ ] Line 85: "facilitate giving" -> "facilitates the giving of"
- [ ] Line 85: "making research more reproducible" -> "making it possible to reproduce the underlying analysis"
- [ ] Line 85: "Ensure that anyone... to do so" -> "You should ensure that anyone who wishes to cite a dataset that you host has access to a persistent identifier in order to do so."
- [ ] Line 102: "to identify" -> "to identify individuals"
- [ ] Line 103: Can you add a comma after "For example"?
- [ ] Line 109: ...and another comma after "Similarly"?

LouiseABowler commented:
Open hardware

- [ ] General point about subtitles: Capitalisation is inconsistent in "Why open hardware?", "Elements of an Open source Hardware Project" and "Open Source Hardware Processes and Practices"
- [ ] Line 119: Two uses of "but" in the same sentence - perhaps have "Whether or not they choose to buy the unit, users can still get a full understanding of how the hardware..." as a standalone sentence
- [ ] Line 123: "equipment's is" -> "equipment is"
- [ ] Line 128: "Cross over" -> "crossover"
- [ ] Line 132: "accoding" -> "according"
- [ ] Line 135: The third level of bullet points seems a lot - would it be possible to cut the "Examples" list out?
- [ ] Line 137: "molded" -> "moulded"
- [ ] Line 142: The abbreviation"OSHW" only gets used once in the chapter, is it worth writing it out in full?
- [ ] General point on "Elements of an Open source Hardware Project": Would this section be better placed in a checklist?

The Open Hardware section feels very specialised and less on-topic than the "Open Source Software" section that comes later on. Would it be possible to swap the two?

LouiseABowler commented:
Open source software

- [ ] Line 182: Cut "that means"
- [ ] Line 182: "your" -> "the" (in the first sentence)
- [ ] Line 182: Add comma after "form"
- [ ] Line 184: Could you add the full name of "NPIY"?
- [ ] Line 189: "people that -> "people who"
- [ ] Line 190: "organizing" -> "organising", "prioritizing" -> "prioritising"
- [ ] Line 203: missing a space before the dash
- [ ] Line 205: "other's" -> "others'"
- [ ] Line 212: "worlds" -> "world's"
- [ ] Line 214: "python" -> "Python"
- [ ] Line 218: "but" -> "put"
- [ ] Line 220: Two spellings of licence/license used here - shall we go for British spelling, or "license" to match the filename recognised by GitHub?
- [ ] Line 246: Could you add a link to Active Admin?
- [ ] Line 289: Missing space before the dash.

LouiseABowler commented:
Open access

- [ ] Lines 321 and 323 : "self archiving" -> "self-archiving"
- [ ] Line 523: Could start with "The aim of the self-archiving movement is to provide ..."
- [ ] Line 325: "preprints servers" -> "preprint servers"
- [ ] Line 340: "license" -> "licence"
- [ ] Line 351: "Self archive" -> "Self-archive"

LouiseABowler commented:
Open notebooks

- [ ] Line 362: "thir: -> "their"

Open scholarship

- [ ] Line 370: "fits" -> "fit"
- [ ] Line 370: Missing space before dash
- [ ] Line 372: What should the link lead to? I'm getting an error when I try to click through
- [ ] Line 376: Missing space before dash
- [ ] Line 376: Could you add a comma after "usage"?
- [ ] Line 382: The formatting for this bulletpoint list isn't displaying properly
- [ ] Line 388: Add a comma after "commons"
- [ ] Line 402: It isn't clear what this link is relevant to unless you click through
- [ ] Line 406: Can you add a comma after "research"?

Checklists

- [ ] Line 421: "license" -> "licence"
- [ ] Line 429: the definition of contrinution guidelines is fairly clear, so you could cut "instructing them on how to go about that"

What to learn next

- [ ] Line 443: Add a comma after "already"

Further reading

- [ ] Could you put the titles of the book/websites into this section?

Definitions/glossary

- [ ] Line 463: "is metadata" -> "are metadata"
- [ ] Lines 465 and 473: "practise" -> "practice"

LouiseABowler commented:
And done! All the comments above are very minor things; the chapter is already in a really good state @r-j-arnold! And I've learnt a lot in the last hour or so :)

r-j-arnold commented:
> Open source software
>
> * [ ]  Line 203: missing a space before the dash

I think this is a stylistic difference, I've most often seen dashes used like this without the front space. I'm ok with changing it if you'd much prefer it though.

> * [x ]  Line 214: "python" -> "Python"
Also changed linux to Linux

> * [ ]  Line 220: Two spellings of licence/license used here - shall we go for British spelling, or "license" to match the filename recognised by GitHub?

I've gone with the UK spelling.

r-j-arnold commented:
> Open scholarship
> 
> * [ ]  Line 372: What should the link lead to? I'm getting an error when I try to click through

It's supposed to display a figure. I'd made a typo in the filename. Should display now.

> * [x ]  Line 382: The formatting for this bulletpoint list isn't displaying properly

Should be fixed now, but just check?

> * [ x]  Line 402: It isn't clear what this link is relevant to unless you click through

This was supposed to be in the bibliography as material used but I missed it. Moved it there now.

> * [ ]  Line 463: "is metadata" -> "are metadata"

"For example (35, 33, 27, 30, 33) is data but the units (miles per hour) and the fact these are the speeds of cars on a certain stretch of road is metadata". It's the *fact* that's the metadata and that's only one single thing so I believe it should be singular?

r-j-arnold commented:
Thanks so much for such thorough reviews and catching so many little details I missed @LouiseABowler and @pherterich! There's a handful I've responded to in comments above where I had a query or a modification to your requested change, let me know if you have issues with any of those.

LouiseABowler commented:
No problem @r-j-arnold - and thanks for putting up with my pickiness! I think everything else you mentioned can be ticked off now, and I noticed a couple of other things - both are stuck in the list below. Go ahead and merge when you're ready (unless anyone else wants more revisions), I won't need to check these points again.

- The images and bulletpoint lists display fine for me now.
- The big list in the open hardware section has got lots of `[]` characters after the bulletpoints - is that an unintended find & replace?
- I'm not too bothered about spaces before dashes; that's what I'd do personally but we should go with your writing style here as you wrote the chapter :)
- For the is/are metadata point, I'd assumed you meant that the units were part of the metadata as well as the fact that the values referred to speeds.
- The last sentence in the further reading section needs revising (or have you not pushed your last changes yet?)

There are a couple of things that might also be worth discussing in relation to the other chapters (don't let these stop you merging this one because they're separate from the content itself):
- What do you think of having a table of contents at the top of each chapter? I'm mentioning this because the open hardware section still feels very long to me, and it would be useful to know if there is more material after that point in the chapter. (In the long-term, formatting the pages nicely would help but we're unlikely to get to that in the next few weeks.)
- Could you add some more description (maybe titles and authors) in the Bibliography section? It looks very brief in the current format.
